utility/mm2_update-master-directory-list

```diff
@@ -52,21 +53,32 @@ class MasterFilter(logging.Filter):
         record.category = cname
         return True
 
-def make_file_details_from_checksums(session, diskpath, relativeDName, D):
+def make_file_details_from_checksums(session, config, D, files=None):
     def _parse_checksum_file(relativeDName, checksumlen):
         r = {}
         try:
-            f = open(os.path.join(diskpath, relativeDName),  'r')
+            f = open(os.path.join(config.get('UMDL_PREFIX', ''), relativeDName),  'r')
             for line in f:
                 line = line.strip()
                 s = line.split()
```

Comment:
This whole block might be easier to read if you did some tuple unpacking here, something like:
```
try:
     field1, field2 = line.split()
     field3, field4 = None, None
except ValueError:
     # Needs more than 2 values to unpack the tuple
    field1, field2, field3, field4 = line.split()  # If this fails it's an unexpected format that results in a ValueError
```
Obviously with more enlightening field names, but you get the idea. It's up to you.